wordpress-mobile/WordPress-iOS
# Stats Refresh: create initial view

Fixes #10291 

This PR contains:
- A `statsRefresh` Feature Flag.
- A new View Controller (`SiteStatsDashboardViewController`) and storyboard (`SiteStatsDashboard`) that is launched when `Stats` is selected and the Feature Flag is enabled. 

The new `SiteStatsDashboardViewController` does the following:
- When `Insights` is selected, displays an empty table via an embedded view.
- When a date filter is selected, displays a just as empty but different table via an embedded view.
- Displays filters using a `FilterTabBar`.
- Has support for a loading `UIProgressView` (that obviously does nothing right now).
- Saves and loads the selected filter to User Defaults.

For clarity, since several of the stats files have similar names:
- Stats is launched from `StatsViewController.m`.
- The new `SiteStatsDashboardViewController.swift` is meant to eventually replace `WPStatsViewController.m`.
- The new `SiteStatsDashboard.storyboard` is meant to eventually replace `SiteStats.storyboard`.

To test:

---
- Launch the app.
- Go to Stats.
- Verify the view is:

![new_stats](https://user-images.githubusercontent.com/1816888/47528959-bd0e6800-d863-11e8-9354-63bc1e255770.png)

---
- Select a filter other than `Insights` (so you can easily see this works).
- Relaunch the app.
- Verify the selected filter matches your selection.
  - NOTE: The same UD key is used as the original Stats, so any selected filter should be reflected in both Stats views (old and new).

---
- Disable the `statsRefresh` Feature Flag.
- Launch the app, and go to Stats.
- Verify the old Stats view appears.
